src/runtime/buffer.c

```diff
@@ -17,6 +17,24 @@ buffer allocate_buffer(heap h, bytes s)
     }
     return b;
 }
+KLIB_EXPORT(allocate_buffer);
+
+void buffer_write(buffer b, const void *source, bytes length)
```

Comment:
Maybe it isn't significant here, but in general taking away inlined functions in data structure code could cause performance-related side-effects. What we could do for these is leave the inlines as-is, make kern_* wrappers like you did for register_timer, but add a macro like KLIB_EXPORT_RENAME to specify a string to use for the exported symbol name, so that they may be exported under their original names - buffer_write, buffer_read, etc.

Reply:
Ok, I added the KLIB_EXPORT_RENAME macro, and created kern_ wrappers for the inline buffer_read and buffer_write functions